qiskit/circuit/library/arithmetic/adders/adder.py

```diff
@@ -39,6 +42,16 @@ class Adder(QuantumCircuit):
 
     """
 
+    @deprecate_func(
+        since="1.3",
+        additional_msg=(
+            "Use the adder gates provided in qiskit.circuit.library.arithmetic instead. "
+            "The gate type depends on the adder kind: fixed, half, full are represented by "
+            "ModularAdderGate, AdderGate, FullAdderGate, respectively. For different adder "
+            "implementations, see https://docs.quantum.ibm.com/api/qiskit/synthesis.",
+        ),
+        pending=True,
+    )
```

Comment:
With the naming conventions introduced in this PR, "Adder" stands for (what used to be called) a "half"-adder, "ModularAdder" stands for a "fixed"-adder, and a "FullAdder" stands for a "full"-adder. Personally, I like these new names, especially "modular" instead of "fixed". My only question is whether we should use the name "Adder" or the name "HalfAdder" to represent an adder without a `c_in`. I think "HalfAdder" is a more standard terminology, and at least to me the name "Adder" can refer to all kinds of adders, not necessarily to the "half"-one. But I am fine with any decision here.

Reply:
An alternative implementation would be to have a single "AdderGate" which internally stores the type of the adder (half/full/modular) and have adder gate plugins take this into account. I am just pointing this out but I prefer the approach taken in the PR.

Reply:
I agree with @alexanderivrii . Perhaps it's worth to have only a single `AddedGate`, with various synthesis methods in the synthesis library and several variants that can be used via the HLS plugin (similar to a single `MCXGate`)

Reply:
Wait, @ShellyGarion, how do you agree with me when I said that I prefer the approach in the PR, except with a "HalfAdder" instead of "Adder"? 😛

Reply:
I agreed with your "alternative implementation" suggested above. But if both you and Julien prefer to have several adders, it's OK with me.

Reply:
Since Modular/Half/Full adder vary in the number of qubits in their mathematical definition, they cannot be represented by the same gate object -- otherwise, where would you apply it on? 🙂 

I'm fine re-re-naming to `HalfAdder` if you prefer 👍🏻

Reply:
For the MCXGate we managed to deal with various methods with different number of qubits (although there these were ancilla qubits). Is it possible to do something similar in this case as well ?

Reply:
Yeah, I like the "HalfAdder" a bit more, and I think "HalfAdder" is the more standard terminology for this circuit: [link to wiki](https://en.wikipedia.org/wiki/Adder_(electronics))

Reply:
@ShellyGarion: The _mathematical definition_ of MCX always has the same number of qubits; we can only deal with varying auxiliary qubits in the _decomposition_. In the adders, the mathematical definition varies, so they cannot be represented by the same gate, because in the circuit modernization `gate == mathematical operation`. 🙂

Reply:
@alexanderivrii: renamed in a54907eec954f98f61bf723fe0e82d5f4d0fb00e